test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,186 @@
+                                   {ATTR_rescavail + '.vntype':
+                                    'cray_compute'})
+        self.assertEqual(nodes_up_apstat,
+                         len(nodes[ATTR_rescavail + '.vntype=cray_compute']))
+
+        nodes = self.server.status(NODE)
+        for node in nodes:
+            self.assertEqual(node['state'], 'free')
+            self.assertEqual(node['resources_assigned.ncpus'], '0')
+            self.assertEqual(node['resources_assigned.mem'], '0kb')
+
+    @tags('cray', 'smoke')
+    def test_cray_login_job(self):
+        """
+        Submit a simple sleep job that requests to run on a login node
+        and expect that job to go in running state on a login node.
+        Verify that the job runs to completion and check job output/error.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_login',
+                             ATTR_N: 'cray_login'})
+
+        scr = []
+        scr += ['echo Hello World\n']
+        scr += ['sleep 5\n']
+
+        sub_dir = self.du.mkdtemp(uid=TEST_USER.uid)
+        j1.create_script(scr)
+        jid1 = self.server.submit(j1, submit_dir=sub_dir)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a login node
+        self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = j1.get_vnodes()[0]
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_login'},
+                           id=vname)
+
+        cu = CrayUtils()
+        # Check if number of compute nodes in use are 0
+        self.assertEqual(cu.count_node_in_state('use'), 0)
+
+        # verify the contents of output/error files
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid1, extend='x')
+        error_file = os.path.join(sub_dir, 'cray_login.e' + jid1.split('.')[0])
+        self.assertEqual(os.stat(error_file).st_size, 0,
+                         msg="Job error file should be empty")
+
+        output_file = os.path.join(
+            sub_dir, 'cray_login.o' + jid1.split('.')[0])
+        with open(output_file, 'r') as of:
+            for line in of:
+                self.assertTrue("Hello World" in line,
+                                msg="Job output file incorrect")
+                break
+
+    @tags('cray', 'smoke')
+    def test_cray_compute_job(self):
+        """
+        Submit a simple sleep job that runs on a compute node and
+        expect the job to go in running state on a compute node.
+        Verify that the job runs to completion and check job output/error.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_compute',
+                             ATTR_N: 'cray_compute'})
```

Comment:
Since mpp* (mppwidth, mppdepth, etc.) resources are deprecated, but still supported, should we have another test using mpp* resources?

Reply:
No I don't think we should have another test using mpp* resources. When writing a new test we won't be using mpp* resources. Also we don't need automate old tests that use a deprecated feature.